client/src/js/services/TrialBalanceService.js

```diff
@@ -163,5 +147,28 @@ function JournalPostingModalService(util, $http, Modal) {
     return $http.post(url, {transactions : transactions});
   }
 
+  /**
+   * @function getRelatedTransaction
+   * @description
+   * This function gets an account_id with a list of transaction and send back
+   * a list of transaction related to this account
+   **/
+  function getRelatedTransaction(accountId, records){
+    var transactions = [], processed = [];
+
+    records.forEach(function (item) {
+      if(processed.indexOf(item.trans_id) === -1 && item.account_id === accountId){
+        transactions = transactions.concat(getTransaction(item.trans_id, records));
+      }
+    });
+    return transactions;
+  }
+
+  function getTransaction(transId, records){
```

Comment:
This will make an `O(n^2)` operation.  If we have 1,000 rows, we will do 1,000,000 lookups.  We can do better.

```js
function getRelatedTransaction(accountId, records) {

  // these are transaction ids of transactions that we will keep.
  var transIds = [];

  // loop through all records, adding related rows to the 
  records.forEach(function (row) {
    
     // determine if we need to keep the transaction by checking the account id
     var isRelated = (row.account_id === accountId);
     if (isRelated) {
        transIds.push(row.trans_id);
     }
  });

  // filter on the transactions
  var transactions = records.filter(function (row) {
    return transIds.indexOf(row.trans_id) > -1;
  });

  return transactions;
}
```

This function now does `O(n)` work (it technically does `O(2n)`) which is an improvement.

You can actually do this in `O(n) + m`, where `m` is the number of affected transactions by indexing first on `trans_id` and then using the index to look up affected rows.  But I think the code would be more complicated than necessary this this operation.

I would actually write the `isRelated` check as a reduce, since it is clearer to me.  Like  this:
```js
  // loop through all records, adding related rows to the 
  var transIds = records.reduce(function (ids, row) {
    var isRelated = (row.account_id === accountId);
    return isRelated ? ids : ids.concat(row);
  } []); 
```

But that is your choice.

For reference, as of Nov 1st, the IMCK database has 1702 unposted rows in the posting journal.